src/rider/main/kotlin/com/jetbrains/rider/plugins/efcore/features/dbcontext/scaffold/ScaffoldDbContextDataContext.kt

```diff
@@ -79,6 +78,13 @@ class ScaffoldDbContextDataContext(intellijProject: Project) : CommonDataContext
         commonDialogState.get(KnownStateKeys.DB_CONTEXT_FOLDER)?.apply {
             dbContextFolder.value = this
         }
+
+        commonDialogState.get(KnownStateKeys.TABLES)?.apply {
+            // We assign the value directly as a listOf() because using tablesList.removeAll() causes the menu to not open at all
```

Comment:
I tried investigating the issue but couldn't figure out much, so I had to do that. I think I'll need your help on this one, to figure out why `.removeAll()` wasn't properly working.

Reply:
Okay, no problem, I'll take a look. If possible, please provide minimal instructions on how to break the dialog with that.

Reply:
> Okay, no problem, I'll take a look. If possible, please provide minimal instructions on how to break the dialog with that.

Replace the line which has `tablesList.value = mutableListOf()`
With a `tablesList.removeAll()` and you will see that it no longer opens the dialog.

Reply:
BTW, I don't think there will ever be a scenario in which the loadState would be called at the tables/schemas tab. This means that there is no way to have an outdated view of the `tablesList`, based on the dialog state.

Reply:
You are correct, in such a scenario it will work without issue. Just trying to keep the observables-based design in sync across the codebase to be predictable.

Reply:
It will also simplify the transition to [`rd`](https://github.com/JetBrains/rd-based `IViewableProperty` in the future, which is used across almost all Rider reactive UI forms and dialogs.

Reply:
Hi! Regarding that - could you try to apply this patch and once again try to use `.clear()` here instead of reassigning the value? Seems it was a bug in a binding logic that caused concurrent modification of `items` list inside `ObservableCollection`.

[Observables__Fix.patch](https://github.com/JetBrains/rider-efcore/files/12166858/Observables__Fix_.addAll._and_.removeAll._causing_.ConcurrentModificationException.patch)